pkg/plugins/runtime/k8s/webhooks/secret_validator.go

```diff
@@ -49,7 +63,22 @@ func (v *SecretValidator) Handle(ctx context.Context, req admission.Request) adm
 		}
 		return admission.Denied(err.Error())
 	}
+	return admission.Allowed("")
+}
 
+func (v *SecretValidator) handleDelete(ctx context.Context, req admission.Request) admission.Response {
+	secret := &kube_core.Secret{}
+	if err := v.Client.Get(ctx, kube_types.NamespacedName{Name: req.Name, Namespace: req.Namespace}, secret); err != nil {
```

Comment:
Isn't Secret a part of `admission.Request`? Can we use it instead of getting a new one from k8s?

Reply:
Unfortunately on `delete` you only have ID of the object: name and namespace.

Reply:
Are there any guarantees that Secret still in k8s store?

Reply:
Good point 👍  there is no guaranteed. I changed the webhook to ignore such case, so standard K8S mechanism will handle this